src/QMCWaveFunctions/TWFPrototype.cpp

```diff
@@ -0,0 +1,316 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2021 QMCPACK developers.
+//
+// File developed by:   Raymond Clay III, rclay@sandia.gov, Sandia National Laboratories
+//
+// File created by:   Raymond Clay III, rclay@sandia.gov, Sandia National Laboratories
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "QMCWaveFunctions/TWFPrototype.h"
+#include "Numerics/DeterminantOperators.h"
+#include <iostream>
+namespace qmcplusplus
+{
+TWFPrototype::TWFPrototype() : initialized(false) { std::cout << "TWFPrototype Constructed\n"; }
+
+TWFPrototype::IndexType TWFPrototype::get_group_index(const IndexType gid)
+{
+  IndexType ndets = groups.size();
+  for (IndexType i = 0; i < ndets; i++)
+    if (gid == groups[i])
```

Comment:
So since you wrote no docs it took awhile to figure out that since addDetermimant might be called out of group order you need this TWFP index to groups id.  Seems like it would be better to just construct the object with a list of determinants sorted in groupid order and reduce internal state.

Reply:
What's current best practices for documentation?  Are you talking about more ample source code comments?  A write up in the manual?  Something else?

Reply:
I thought to point to the manual (https://qmcpack.readthedocs.io/en/develop/developing.html#comments) on comments here since a response was not forthcoming.  Upon looking at the code it seems like there is extensive documentation in the header file for TWFPrototype for both the class and the functions.  The only thing that seemed missing to me was labeling parameters [in,out], but Peter seems to be suggesting something broader.  @PDoakORNL , what are you looking for?